src/objects/Mesh.js

```diff
@@ -147,17 +147,48 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 		}
 
-		function checkIntersection( object, material, raycaster, ray, pA, pB, pC, point ) {
+		function checkIntersection( object, material, raycaster, ray, pA, pB, pC, point, faceIndex ) {
 
 			var intersect;
+			var faceMaterial;
+			
+			if ( material.length && faceIndex ) {
+				
+				var groups = object.geometry.groups;
+				
+				if ( groups ) {
+					
+					for ( var i = 0; i < groups.length; i++ ) {
+						
+						var group = groups[ i ];
+						
+						if ( faceIndex >= group.start && faceIndex <= group.start + group.count ) {
```

Comment:
@Mugen87 Is this right?   `faceIndex` with groups?   `<=`?

And shouldn't the material be determined outside this function to prevent iterating through this loop each time?